src/openforms/js/components/admin/form_design/utils.js

```diff
@@ -57,6 +57,74 @@ const checkKeyChange = (mutationType, newComponent, oldComponent) => {
   return newComponent.key !== oldComponent.key;
 };
 
+const checkIfInitialValueTransformationIsNeeded = (newType, originalVariable) => {
+  // If the initialValue is null/undefined, we should transform
+  if (originalVariable?.initialValue == null) {
+    return true;
+  }
+
+  switch (newType) {
+    case 'array':
+      // If the value isn't in array format, it should be transformd
+      return !Array.isArray(originalVariable.initialValue);
+
+    case 'boolean':
+      return typeof originalVariable.initialValue !== 'boolean';
+
+    case 'object':
+      return typeof originalVariable.initialValue !== 'object';
+
+    case 'int':
+    case 'float':
+      return typeof originalVariable.initialValue !== 'number';
+
+    case 'string':
+    case 'datetime':
+    case 'date':
+    case 'time':
+      return typeof originalVariable.initialValue !== 'string';
+
+    default:
+      return false;
+  }
+};
+
+const transformInitialValue = (newType, originalVariable) => {
+  switch (newType) {
+    case 'array':
+      if (
+        originalVariable?.initialValue == null ||
+        typeof originalVariable.initialValue === 'object'
```

Comment:
`Array` is also object so I'm not sure how robust this is.

Reply:
Ah yeah, it should have a `Array.isArray(...)` to make sure..